### ISSUE TITLE
Directories with many subdirectories are slow to iterate because of inlined set operation

### ISSUE THREAD

**Issue by chroth** [jaraco/zipp#43]
After upgrading zipp from version 2.0.1 to 2.2.0 some tests in a project I'm working on started running at ~3x the speed. The slowness came from zipp iterating the pyspark directory. When I started looking into to why, I was able to narrow it down to this commit: https://github.com/jaraco/zipp/commit/0f036c6406052426019f32c85d882e44903be89f

Reproduce by timing the following code with and without the set operation inlined:
```python
data = ["x/%s" % i for i in range(10000)]
zipp.CompleteDirs._implied_dirs(data)
```

With `set(names)` inlined for _me_ the above code took on average ~90x longer to run with one "sub directory", ~110x with 2, ~120x with 3. So I'd assume that directories with many subdirectories, especially with some nesting are slow.

Tested on:
Python 3.6.5 and 3.6.9
Zipp version: 2.1.0+

**Comment by jaraco** [comment]
Gah, yeah, you're right. I was annoyed by the variable named 'name_set' and when I went to address that concern, I removed the performance optimization that variable was intending to have.

**Comment by jaraco** [comment]
Fixed in v1.1.1 and v2.1.1, including a test to prevent regression.
